### PR TITLE
Use OS-specific terminology for "Move to Trash" in the editor filesystem

### LINKED ISSUE #2460: Rename 'Move to Trash' to 'Move to Recycle bin' for Windows
### Describe the project you are working on
Small testing of Godot

### Describe the problem or limitation you are having in your project
Currently on 3.2.4 rc 'Move to Trash' is the action when you 'delete' a file from the file browser. What is Trash on Windows?

### Describe the feature / enhancement and how it helps to overcome the problem or limitation
On windows 'Trash' is not recognised by many users. Recycle bin is standard. 
Users (beginners mainly) may think 'Is this a Godot specific folder' or 'where are my files?', not knowing the recycle bin is the pseudonym for 'Trash'. It's non-obvious.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
Rename 'Trash' to 'Recycle Bin'

### If this enhancement will not be used often, can it be worked around with a few lines of script?
No
### Is there a reason why this should be core and not an add-on in the asset library?
Core

### PR DESCRIPTION
**Pull request by Calinou**
This closes https://github.com/godotengine/godot-proposals/issues/2460.

### CODE REVIEW

File: editor/filesystem_dock.cpp [hunk 1]
```diff
@@ -2718,7 +2718,14 @@ FileSystemDock::FileSystemDock(EditorNode *p_editor) {
 	// `KEY_MASK_CMD | KEY_C` conflicts with other editor shortcuts.
 	ED_SHORTCUT("filesystem_dock/copy_path", TTR("Copy Path"), KEY_MASK_CMD | KEY_MASK_SHIFT | KEY_C);
 	ED_SHORTCUT("filesystem_dock/duplicate", TTR("Duplicate..."), KEY_MASK_CMD | KEY_D);
+#ifdef WINDOWS_ENABLED
+	// TRANSLATORS: This string is only used on Windows, as it refers to the system trash
+	// as "Recycle Bin" instead of "Trash". Make sure to use the translation of "Recycle Bin"
+	// recommended by Microsoft for the target language.
+	ED_SHORTCUT("filesystem_dock/delete", TTR("Move to Recycle Bin"), KEY_DELETE);
+#else
```
Comment (bruvzg) [inline editor/filesystem_dock.cpp:2726]: If we all platform specific text, on macOS it should be `Move to Bin`.

File: editor/filesystem_dock.cpp [hunk 2]
```diff
@@ -2718,7 +2718,24 @@ FileSystemDock::FileSystemDock(EditorNode *p_editor) {
 	// `KEY_MASK_CMD | KEY_C` conflicts with other editor shortcuts.
 	ED_SHORTCUT("filesystem_dock/copy_path", TTR("Copy Path"), KEY_MASK_CMD | KEY_MASK_SHIFT | KEY_C);
 	ED_SHORTCUT("filesystem_dock/duplicate", TTR("Duplicate..."), KEY_MASK_CMD | KEY_D);
+
+#ifdef WINDOWS_ENABLED
+	// TRANSLATORS: This string is only used on Windows, as it refers to the system trash
+	// as "Recycle Bin" instead of "Trash". Make sure to use the translation of "Recycle Bin"
+	// recommended by Microsoft for the target language.
+	ED_SHORTCUT("filesystem_dock/delete", TTR("Move to Recycle Bin"), KEY_DELETE);
+#else
```
Comment (akien-mga) [inline editor/filesystem_dock.cpp:2727]: Use `#elif defined()` instead.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.4.

**Comment by akien-mga** [comment]
> Cherry-picked for 3.2.4.

Undid cherry-pick as the change doesn't seem consensual for now (see https://github.com/godotengine/godot-proposals/issues/2463, even if it's more about the *previous* PR that changed this), and it would require translation updates close to the release.

**Comment by akien-mga** [comment]
For the reference, this was reverted by #49519 while renaming it to "Delete" for all platforms.

**Pull request merged**